src/bson_mapper/stdx/optional.hpp

```diff
@@ -0,0 +1,26 @@
+// Copyright 2016 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <bsoncxx/stdx/optional.hpp>
+
+namespace bson_mapper {
+namespace stdx {
+
+using bsoncxx::stdx::optional;
+using bsoncxx::stdx::nullopt;
+using bsoncxx::stdx::make_optional;
+}  // namespace stdx
+}  // namespace stdx
```

Comment:
They were previously in the `bsoncxx::` namespace which I think would be kind of user unfriendly for a potential odm user.